python/lsst/ts/athexapod/controller.py

```diff
@@ -129,33 +116,27 @@ async def write_command(self, cmd, has_response=True, num_line=1):
             List with the response(s) from the command.
 
         """
-
-        async with self.lock:
-            if not self.is_connected:
-                raise RuntimeError(
-                    "Not connected to hexapod controller. " "Call `connect` first"
-                )
-
-            self.log.debug(f"Writing: {cmd.encode()!r}")
-            self.writer.write(f"{cmd}\n".encode())
-            await self.writer.drain()
-
-            if has_response:
-                try:
-                    replies = []
-                    for i in range(num_line):
-                        raw_line = await self.reader.readuntil(b"\n")
-                        line = raw_line.decode("ISO-8859-1").strip()
-                        self.log.debug(f"Read {i+1} of {num_line} lines: {line}")
-                        replies.append(line)
-                except asyncio.TimeoutError:
-                    self.log.warning(
-                        "Timed out waiting for response from controller. Result "
-                        "may be incomplete."
-                    )
-                    raise
-
-                return replies
+        if self.client.connected:
+            async with self.lock:
```

Comment:
You have removed the use of this lock everywhere else so remove it here as well. And remove it from `__init__`.

Reply:
But it literally is the only line that uses the lock so it is redundant and you should remove it.